### DISCUSSION TITLE
Routing between agents

### DISCUSSION THREAD

**Discussion by libin85** [microsoft/autogen#1011] [Q&A] [answered]
Hi,

I've successfully developed a chatbot using Langchain that efficiently handles user queries about a specific system. Now, I aim to enhance this by creating a multi-agent system utilizing AutoGen, where each agent specializes in a distinct area. An administrative agent would determine and direct the query to the most suitable agent for resolution. I'm considering using GPT to facilitate this routing process. The idea is to input a prompt into GPT that describes the query and lists the available agents, asking GPT to identify which agent is best suited to address the query. Do you have any suggestions for improving this query-routing mechanism?

**Comment by PyroGenesis** [comment]
I had a similar use case as you, and I also created an administrative agent (Commander) to control the flow of the conversation. 

I used [this notebook](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_hierarchy_flow_using_select_speaker.ipynb) as a reference and my implementation ended up looking [like so](https://github.com/microsoft/autogen/issues/613#issuecomment-1813516949). Later, I connected the Commander to a vector DB that would add past prompts and correct responses as context to improve consistency (in lieu of fine-tuning).

**Comment by rickyloynd-microsoft** [comment]
@afourney

**Comment by afourney** [comment, marked as answer]
What you describe is essentially what the GroupChat already does (if you limit GroupChat to 1 turn). Have a look and see if it meets your needs. Note that GroupChat now uses the agent.description field: https://microsoft.github.io/autogen/blog/2023/12/29/AgentDescriptions
